lib/webpacker/compiler.rb

```diff
@@ -13,13 +13,17 @@ def initialize(webpacker)
 
   def compile
     if stale?
+      clobber
       record_compilation_timestamp
       run_webpack
-    else
-      logger.debug "No compiling needed as everything is fresh"
     end
   end
 
+  def clobber
+    config.public_output_path.rmtree if config.public_output_path.exist?
+    config.cache_path.rmtree if config.cache_path.exist?
+  end
```

Comment:
Actually, why are we automatically clobbering on every compile? I don't think Sprockets does that? In production, you may well want to keep older assets around for a bit such that you don't break in-flight requests during a deploy.